cmd/cluster/aws/destroy.go

```diff
@@ -138,20 +138,26 @@ func DestroyCluster(ctx context.Context, o *core.DestroyOptions) error {
 	return core.DestroyCluster(ctx, hostedCluster, o, destroyPlatformSpecifics)
 }
 
-// ValidateCredentialInfo validates if the credentials secret name is empty, the aws-creds is not empty; validates if
+// ValidateCredentialInfo validates if the credentials secret name is empty, the aws-creds or sts-creds mutually exclusive and are not empty; validates if
 // the credentials secret is not empty, that it can be retrieved.
 func ValidateCredentialInfo(opts *core.DestroyOptions) error {
 	if len(opts.CredentialSecretName) == 0 {
-		if err := IsRequiredOption("aws-creds", opts.AWSPlatform.AWSCredentialsFile); err != nil {
+		if err := opts.AWSPlatform.AWSCredentialsOpts.Validate(); err != nil {
 			return err
 		}
-	} else {
-		// Check the secret exists now, otherwise stop
-		opts.Log.Info("Retrieving credentials secret", "namespace", opts.Namespace, "name", opts.CredentialSecretName)
-		if _, err := util.GetSecret(opts.CredentialSecretName, opts.Namespace); err != nil {
+		return nil
+	}
+
+	if opts.AWSPlatform.AWSCredentialsOpts.AWSCredentialsFile == "" {
```

Comment:
Is there a reason we're doing this check again? Wouldn't Validate above catch it?

Reply:
Validate above is executed when the CredentialsSecret is not set.

If CredentialsSecret is set we want to only check that `role-arn` is required, but only if AWSCredentialsFile is not set. i.e. AWSCredentialsFile  has priority over credentialSecret (this is the current behavior, I can change it if it doesn't make sense)